src/base/static/state/ducks/map.js

```diff
@@ -0,0 +1,110 @@
+// Selectors
+export const mapLayersBasemapSelector = state => {
+  return {
+    visibleBasemapId: state.mapLayers.visibleBasemapId,
+    priorVisibleBasemapId: state.mapLayers.priorVisibleBasemapId,
+  };
+};
+export const mapSizeValiditySelector = state => {
+  return state.mapLayers.isValidSize;
+};
+export const mapPositionSelector = state => {
+  return state.mapLayers.mapPosition;
+};
+
+// Actions
+const TOGGLE_LAYER_VISIBILITY = "map/TOGGLE_LAYER_VISIBILITY";
+const SET_BASEMAP = "map/SET_BASEMAP";
+const SET_MAP_POSITION = "map/SET_MAP_POSITION";
+const SET_MAP_SIZE_VALIDITY = "map/SET_MAP_SIZE_VALIDITY";
+
+// Action creators
+export const toggleLayerVisibility = layerId => {
+  return {
+    type: TOGGLE_LAYER_VISIBILITY,
+    payload: {
+      layerId: layerId,
+    },
+  };
+};
+export const setBasemap = layerId => {
+  return {
+    type: SET_BASEMAP,
+    payload: {
+      basemapId: layerId,
+    },
+  };
+};
+export const setMapPosition = mapPosition => {
+  return {
+    type: SET_MAP_POSITION,
+    payload: mapPosition,
+  };
+};
+export const setMapSizeValidity = isValidSize => {
+  return {
+    type: SET_MAP_SIZE_VALIDITY,
+    payload: isValidSize,
+  };
+};
+
+// Reducers
+const INITIAL_STATE = {
+  priorVisibleBasemapId: undefined,
```

Comment:
What's the point of tracking this? It's a code smell, and doesn't seem to get used.